tests/helper_functions.py

```diff
@@ -28,33 +35,75 @@ def score_model_in_sagemaker_docker_container(model_path, data):
     """
     env = dict(os.environ)
     env.update(LC_ALL="en_US.UTF-8", LANG="en_US.UTF-8")
-    proc = Popen(['mlflow', 'sagemaker', 'run-local', '-m', model_path], stdout=PIPE, stderr=STDOUT,
+    proc = Popen(['mlflow', 'sagemaker', 'run-local', '-m', model_path, '-p', "5000"],
+                 stdout=PIPE,
+                 stderr=STDOUT,
                  universal_newlines=True, env=env)
+    r = _score_proc(proc, 5000, data, "json").content
+    import json
+    return json.loads(r)  # TODO: we should return pd.Dataframe the same as pyfunc serve
+
+
+def pyfunc_serve_and_score_model(model_path, data):
+    env = dict(os.environ)
+    env.update(LC_ALL="en_US.UTF-8", LANG="en_US.UTF-8")
+    cmd = ['mlflow', 'pyfunc', 'serve', '-m', model_path, "-p", "0"]
+    proc = Popen(cmd,
+                 stdout=PIPE,
+                 stderr=STDOUT,
+                 universal_newlines=True,
+                 env=env)
+    for x in iter(proc.stdout.readline, ""):
+        print(x)
+        m = re.match(pattern=".*Running on http://127.0.0.1:(\\d+).*", string=x)
```

Comment:
The regular expression match is there to read the port number, I start it at port 0 which lets os pick whatever port is available and I than read it from the output.

I agree that  the message can change, but that is a) unlikely, b) simple to fix. 

I could change the code to run on a set port similar to sagemaker scoring but I prefer this way so that users can run this test on their laptop and not interfere with other stuff that might be running.

Reply:
Discussed offline. This seems fine.